File: src/main/scala/robosky/uplands/entity/EntityRegistry.scala
package robosky.uplands.entity

import net.fabricmc.fabric.api.entity.FabricEntityTypeBuilder

import net.minecraft.entity.{EntityCategory, EntityDimensions, EntityType}
import net.minecraft.item.{Item, ItemGroup, SpawnEggItem}
import net.minecraft.util.registry.Registry
import net.minecraft.world.World

import robosky.uplands.UplandsMod

object EntityRegistry {

  val HEXAHAEN: EntityType[HexahaenEntity] = FabricEntityTypeBuilder
    .create(EntityCategory.MONSTER, new HexahaenEntity(_, _))
    .size(EntityDimensions.fixed(1.25f, 1.25f))
    .build

  private def registerMob(name: String, entityType: EntityType[_], primaryColor: Int, secondaryColor: Int): Unit = {
    Registry.register(Registry.ENTITY_TYPE, UplandsMod :/ name, entityType)
    Registry.register(Registry.ITEM, UplandsMod :/ (name + "_spawn_egg"),
        new SpawnEggItem(entityType, primaryColor, secondaryColor, new Item.Settings().group(ItemGroup.MISC)))
  }

  def init(): Unit = {
    registerMob("hexahaen", HEXAHAEN, 0x0e7543, 0xeb8154)
  }
}

File: src/main/java/robosky/uplands/entity/EntityRegistry.java
package robosky.uplands.entity;

import robosky.uplands.UplandsMod;

import net.fabricmc.fabric.api.object.builder.v1.entity.FabricDefaultAttributeRegistry;
import net.fabricmc.fabric.api.object.builder.v1.entity.FabricEntityTypeBuilder;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityDimensions;
import net.minecraft.entity.EntityType;
import net.minecraft.entity.SpawnGroup;
import net.minecraft.entity.mob.HostileEntity;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.SpawnEggItem;
import net.minecraft.util.registry.Registry;

public class EntityRegistry {

    public static final EntityType<HexahaenEntity> HEXAHAEN = FabricEntityTypeBuilder
        .create(SpawnGroup.MONSTER, (EntityType.EntityFactory<HexahaenEntity>)HexahaenEntity::new)
        .dimensions(EntityDimensions.fixed(1.25f, 1.25f)).build();

    public static void init() {
        registerMob("hexahaen", HEXAHAEN, 0x0e7543, 0xeb8154);
        FabricDefaultAttributeRegistry.register(HEXAHAEN, HostileEntity.createHostileAttributes());
    }

    private static <E extends Entity> void registerMob(String name, EntityType<E> entityType, int primaryColor, int secondaryColor) {
        Registry.register(Registry.ENTITY_TYPE, UplandsMod.id(name), entityType);
        Registry.register(Registry.ITEM, UplandsMod.id(name + "_spawn_egg"),
            new SpawnEggItem(entityType, primaryColor, secondaryColor, new Item.Settings().group(ItemGroup.MISC)));
    }
}
